### PR TITLE
RFC: Ginkgo: First bootstrap of using BeforeAll and AfterAll

### PR DESCRIPTION
**Pull request by eloycoto**
This commit fixed our outstanding issue about creating and deleting containers
for the same Testcase and do not have the option for setup and teardown
actions.

This code added a new vendor dependency, "ginkgo-ext" that it's a copy
of `Mirantis/Virtlet` e2e ginkgo-extension with a few additions to
support `--focus` option.

The main changes in this code are:

- `BeforeAll()` replaces BeforeSuite. On the other hand, `BeforeAll` can
be used on Describe and Context  like the initialize function that we're
using now.

- `AfterAll()` replaces `AfterSuite()`. On the other hand, `AfterAll()`
can be used in Describe and Context to clean the state after the
testCase finished, like tearDown function in any unit testing function.

The example TestCase can be the following:

```
var _ = Describe("K8Spolicies", func() {

        AfterAll(func() {
                fmt.Println("AfterAll---<")
        })

		AfterEach(func(){
			fmt.Println("AfterEach")
		})

        Context("Test Group", func() {

                AfterAll(func() {
                        fmt.Println("AfterAll---<Context")
                })

                It("Test1", func() {
                        Expect(true).To(BeTrue())
                })

                It("Test2", func() {
                        Expect(true).To(BeTrue())
                })
        })

        It("Test1", func() {
                Expect(true).To(BeTrue())
        })

        It("Test2", func() {
                Expect(true).To(BeTrue())
        })
})
```

ATM I only migrated `RuntimeValidatedMonitor`, where the AfterAll/BeforeEach is
used, the idea is to run the test for a few days, and if nothing fails move to
other testCases.

Pending things: 

- [] Move eloycoto/ginkgo-ext to cilium org.

### CODE REVIEW

**Review by aanm — COMMENTED** [review]
This looks sane to me but why not simply copy the `vendor/github.com/eloycoto/ginkgo-ext/scopes.go` to `/test` directory?

File: test/test_suite_test.go [hunk 1]
```diff
@@ -78,8 +79,10 @@ func TestTest(t *testing.T) {
 	}
 	junitReporter := reporters.NewJUnitReporter(fmt.Sprintf(
 		"%s.xml", ginkgoext.GetScopeWithVersion()))
-	RunSpecsWithDefaultAndCustomReporters(
-		t, ginkgoext.GetScopeWithVersion(), []Reporter{junitReporter})
+	// RunSpecsWithDefaultAndCustomReporters(
+	// 	t, ginkgoext.GetScopeWithVersion(), []ginkgo.Reporter{junitReporter})
+
+	RunSpecsWithDefaultAndCustomReporters(t, "Foo Suite", []ginkgo.Reporter{junitReporter})
```
Comment (ianvernon) [inline test/test_suite_test.go]: What is the need for this change?

**Comment by ianvernon** [comment]
Looks fine, same concerns as @aanm re: the vendoring.

**Comment by ianvernon** [comment]
test-me-please

File: test/test_suite_test.go (hunk 1 above)
Comment (eloycoto) [reply test/test_suite_test.go]: leftover

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
Looks great! Super stoked on this. Filed https://github.com/cilium/cilium/issues/3109 to track the work to migrate all tests to use this new functionality.

**Review by aanm — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
The go build was hang for a while and it didn't install. Just testing again. 
```
/plugins/cilium-docker'
17:31:04     runtime: go build -i -ldflags '-X "github.com/cilium/cilium/pkg/version.Version=1.0.0-rc7 fda5f3a3a 2018-03-12T16:39:22+00:00 go version go1.9.3 linux/amd64"' -tags lockdebug -o cilium-docker
17:31:52     runtime: make[2]: Leaving directory '/home/vagrant/go/src/github.com/cilium/cilium/plugins/cilium-docker'
17:31:52     runtime: make[2]: Entering directory '/home/vagrant/go/src/github.com/cilium/cilium/plugins/cilium-cni'
17:31:52     runtime: go build -i -ldflags '-X "github.com/cilium/cilium/pkg/version.Version=1.0.0-rc7 fda5f3a3a 2018-03-12T16:39:22+00:00 go version go1.9.3 linux/amd64"' -tags lockdebug -o cilium-cni ./cilium-cni.go
18:58:46 Sending interrupt signal to process
18:58:50 Terminated
```

**Pull request merged**